src/components/MobileWalletDialog/routes/DeleteWallet/Confirm.tsx

```diff
@@ -0,0 +1,66 @@
+import { WarningIcon } from '@chakra-ui/icons'
+import { Alert, AlertDescription, Button, Heading, Stack, Text } from '@chakra-ui/react'
+import { useCallback, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { DialogBackButton } from 'components/Modal/components/DialogBackButton'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import { DialogHeader, DialogHeaderLeft } from 'components/Modal/components/DialogHeader'
+import { SlideTransition } from 'components/SlideTransition'
+import { deleteWallet } from 'context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import type { MobileLocationState } from 'context/WalletProvider/MobileWallet/types'
+import { WalletCard } from 'pages/ConnectWallet/components/WalletCard'
+
+type ConfirmDeleteProps = {
+  vault?: MobileLocationState['vault']
```

Comment:
Should this component assume defined `vault` and have its `<DeleteWallet />` do the check?